catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -228,4 +237,24 @@ public boolean equals(Object obj) {
         .append(descriptors, other.getAttributeDescriptors())
         .isEquals();
   }
+
+  private void validateDescriptors() {
+    Map<String, List<AttributeDescriptor>> descriptorMap = new HashMap<>();
+    for (AttributeDescriptor descriptor : descriptors) {
+      if (descriptor == null) {
+        continue;
+      }
+      descriptorMap.putIfAbsent(descriptor.getName(), new ArrayList<>());
```

Comment:
❓ Instead of iterating through this once to add to the map then again to check for duplicates - could we just check if it already exists here and do the logger warn in this initial iteration?

Reply:
Doh! Ya that would be better.

Reply:
I think what happened was originally I wanted one log statement per attribute with all the conflicting descriptors printed out which could have been > 2. That was useful for met to track things down but doesn't make sense now. Fixing